### ISSUE TITLE
Custom webhook url is ignored

### ISSUE THREAD

**Issue by nicolas-t** [mollie/laravel-cashier-mollie#141]
Hello,

My custom `aftercare_webhook_url` config is ignored.

I'm using custom webhooks locally.

My `config/cashier.php` is as follows : 
```php
return [
    'webhook_url' => 'https://ff70-89-156-98-187.ngrok.io/webhooks/mollie',
    'aftercare_webhook_url' => 'https://ff70-89-156-98-187.ngrok.io/webhooks/mollie/aftercare',
    //...
    'first_payment' => [
        'webhook_url' => 'https://ff70-89-156-98-187.ngrok.io/webhooks/mollie/first-payment',
    //...
```
I'm generating a `checkoutUrl` like this : 

```php
$authUser->newSubscription('main', 'my_plan')->create();
return $response->payment()->getCheckoutUrl();
```

After paying; the webhook `POST /webhooks/mollie/first-payment` is called but returns an error 500 : 
```
Error executing API call (422: Unprocessable Entity): The webhook URL is invalid because it is unreachable from Mollie's point of view.. Field: webhookUrl. Documentation: https://docs.mollie.com/reference/v2/payments-api/update-payment {"exception":"[object] (Mollie\\Api\\Exceptions\\ApiException(code: 422): [2022-10-28T19:12:24+0000] Error executing API call (422: Unprocessable Entity): The webhook URL is invalid because it is unreachable from Mollie's point of view.. Field: webhookUrl. Documentation: https://docs.mollie.com/reference/v2/payments-api/update-payment at myProject/vendor/mollie/mollie-api-php/src/Exceptions/ApiException.php:114)
[stacktrace]
#0 myProject/vendor/mollie/mollie-api-php/src/HttpAdapter/Guzzle6And7MollieHttpAdapter.php(175): Mollie\\Api\\Exceptions\\ApiException::createFromResponse()
#1 myProject/vendor/mollie/mollie-api-php/src/HttpAdapter/Guzzle6And7MollieHttpAdapter.php(104): Mollie\\Api\\HttpAdapter\\Guzzle6And7MollieHttpAdapter->parseResponseBody()
#2 myProject/vendor/mollie/mollie-api-php/src/MollieApiClient.php(538): Mollie\\Api\\HttpAdapter\\Guzzle6And7MollieHttpAdapter->send()
#3 myProject/vendor/mollie/mollie-api-php/src/MollieApiClient.php(494): Mollie\\Api\\MollieApiClient->performHttpCallToFullUrl()
#4 myProject/vendor/mollie/mollie-api-php/src/Endpoints/EndpointAbstract.php(101): Mollie\\Api\\MollieApiClient->performHttpCall()
#5 myProject/vendor/mollie/mollie-api-php/src/Endpoints/PaymentEndpoint.php(72): Mollie\\Api\\Endpoints\\EndpointAbstract->rest_update()
#6 myProject/vendor/mollie/mollie-api-php/src/Resources/Payment.php(699): Mollie\\Api\\Endpoints\\PaymentEndpoint->update()
#7 myProject/vendor/mollie/laravel-cashier-mollie/src/Mollie/UpdateMolliePayment.php(13): Mollie\\Api\\Resources\\Payment->update()
#8 myProject/vendor/mollie/laravel-cashier-mollie/src/Http/Controllers/FirstPaymentWebhookController.php(31): Laravel\\Cashier\\Mollie\\UpdateMolliePayment->execute()
#9 myProject/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(48): Laravel\\Cashier\\Http\\Controllers\\FirstPaymentWebhookController->handleWebhook()
```

It seems that the webhook url used here is incorrect (it ignores my custom domain): 
https://github.com/mollie/laravel-cashier-mollie/blob/8c1d5babc3c3e0339da91532983fe0790811ba11/src/Http/Controllers/FirstPaymentWebhookController.php#L28

In my case something like this would make more sense : 
```diff
- $payment->webhookUrl = route('webhooks.mollie.aftercare');
+ $payment->webhookUrl = config('cashier.aftercare_webhook_url');
```

If you confirm that this is not the expected behaviour I can work on a pull request, with some of your guidance.

Thanks

**Comment by ciungulete** [comment]
Hi @nicolas-t,

Can u try next things:
1. remove https://ff70-89-156-98-187.ngrok.io/ from the config file and keep webhooks without full url.
2. In .env file, change the `APP_URL` to https://ff70-89-156-98-187.ngrok.io/ (the accessible app url)
3. Check in the Mollie dashboard, at the payments and look at the payment history in the right column 
<img width="358" alt="screenshot 135" src="https://user-images.githubusercontent.com/742128/199005966-0ddf38e7-b367-4b02-af58-d434587de19b.png">

Let me know if it works.

**Comment by nicolas-t** [comment]
Hello,

Thanks for your answer.

In my project I have 2 route domains: 
- 1 for api routes (http://api.my-project.local)
- 1 for web routes, where mollie webhooks handler seems to be registered (http://my-project.local or https://ff70-89-156-98-187.ngrok.io)


This time it fails when creating the subscription :  
`$authUser->newSubscription('main', 'myProject')->create();`
returns  
```json
{
  "message": "[2022-10-31T12:59:53+0000] Error executing API call (422: Unprocessable Entity): The webhook URL is invalid because it is unreachable from Mollie's point of view.. Field: webhookUrl. Documentation: https://docs.mollie.com/overview/handling-errors",
  "exception": "Mollie\\Api\\Exceptions\\ApiException",
  "file": "my-project/vendor/mollie/mollie-api-php/src/Exceptions/ApiException.php",
  "line": 114,
  "trace": [
    {
      "file": "my-project/vendor/mollie/mollie-api-php/src/HttpAdapter/Guzzle6And7MollieHttpAdapter.php",
      "line": 175,
      "function": "createFromResponse",
      "class": "Mollie\\Api\\Exceptions\\ApiException",
      "type": "::"
    },
    {
      "file": "my-project/vendor/mollie/mollie-api-php/src/HttpAdapter/Guzzle6And7MollieHttpAdapter.php",
      "line": 104,
      "function": "parseResponseBody",
      "class": "Mollie\\Api\\HttpAdapter\\Guzzle6And7MollieHttpAdapter",
      "type": "->"
    },
    {
      "file": "my-project/vendor/mollie/mollie-api-php/src/MollieApiClient.php",
      "line": 538,
      "function": "send",
      "class": "Mollie\\Api\\HttpAdapter\\Guzzle6And7MollieHttpAdapter",
      "type": "->"
    },
    {
      "file": "my-project/vendor/mollie/mollie-api-php/src/MollieApiClient.php",
      "line": 494,
      "function": "performHttpCallToFullUrl",
      "class": "Mollie\\Api\\MollieApiClient",
      "type": "->"
    },
    {
      "file": "my-project/vendor/mollie/mollie-api-php/src/Endpoints/EndpointAbstract.php",
      "line": 76,
      "function": "performHttpCall",
      "class": "Mollie\\Api\\MollieApiClient",
      "type": "->"
    },
    {
      "file": "my-project/vendor/mollie/mollie-api-php/src/Endpoints/PaymentEndpoint.php",
      "line": 52,
      "function": "rest_create",
      "class": "Mollie\\Api\\Endpoints\\EndpointAbstract",
      "type": "->"
    },
    {
      "file": "my-project/vendor/mollie/laravel-cashier-mollie/src/Mollie/CreateMolliePayment.php",
      "line": 14,
      "function": "create",
      "class": "Mollie\\Api\\Endpoints\\PaymentEndpoint",
      "type": "->"
    },
    {
      "file": "my-project/vendor/mollie/laravel-cashier-mollie/src/FirstPayment/FirstPaymentBuilder.php",
      "line": 132,
      "function": "execute",
      "class": "Laravel\\Cashier\\Mollie\\CreateMolliePayment",
      "type": "->"
    },
    {
      "file": "my-project/vendor/mollie/laravel-cashier-mollie/src/SubscriptionBuilder/FirstPaymentSubscriptionBuilder.php",
      "line": 114,
      "function": "create",
      "class": "Laravel\\Cashier\\FirstPayment\\FirstPaymentBuilder",
      "type": "->"
    },
    {
      "file": "my-project/app/Http/Controllers/UserController.php",
      "line": 180,
      "function": "create",
      "class": "Laravel\\Cashier\\SubscriptionBuilder\\FirstPaymentSubscriptionBuilder",
      "type": "->"
    },
    ....
``` 
In fact here 
https://github.com/mollie/laravel-cashier-mollie/blob/8c1d5babc3c3e0339da91532983fe0790811ba11/src/Mollie/CreateMolliePayment.php#L14
`$payload['webhookUrl']` equals  `"http://api.my-project.local/webhooks/mollie/first-payment"`
It is wrong, because it uses the `api` subdomain (webhooks are not registered there), and because it ignores the `APP_URL` config. 

Hope it is clearer, don't hesitate if you have any question.

**Comment by sandervanhooft** [comment]
Hi @nicolas-t ,

Thanks for following up.

In order for us to reproduce this issue:

1. Can you share the actual code that triggers the issue? I.e. `$billable->newSubscription()->...->create();`?

2. If it's related to creating subscriptions, can you share your `cashier_plans.php` config file?

**Comment by sandervanhooft** [comment]
Ah sorry, I misread part of your initial ticket. Skip point 1 above, only point 2 please :)

**Comment by Naoray** [comment]
Hi @nicolas-t,

sorry for taking over a year to answer back on this one. If you are still having issues with your webhook configuration, please let me know.

In general what you reported initially is intended behaviour! The webhook option takes a relative or absolute url and then builds a full url based upon your app’s `APP_URL` setting.

If you want your subdomain to be recognized, you can simply refer to my answer in https://github.com/mollie/laravel-cashier-mollie/issues/177#issuecomment-1604065137

Tl;dr;
- redefine the webhook urls in your own route file with a `domain()` group
- call `Cashier::ignoreRoutes()` in your `AppServiceProvider@register()` method

```php
Route::namespace('\Laravel\Cashier\Http\Controllers')->domain(‘your_subdomain’)->group(function () {
    Route::name('webhooks.mollie.default')->post(
        Cashier::webhookUrl(),
        [WebhookController::class, 'handleWebhook']
    );

    Route::name('webhooks.mollie.aftercare')->post(
        Cashier::aftercareWebhookUrl(),
        [AftercareWebhookController::class, 'handleWebhook']
    );

    Route::name('webhooks.mollie.first_payment')->post(
        Cashier::firstPaymentWebhookUrl(),
        [FirstPaymentWebhookController::class, 'handleWebhook']
    );
});
```
